7_deploy_osh/roles/suse-build-images/tasks/main.yml

```diff
@@ -0,0 +1,88 @@
+---
+
+- name: Install system packages for build images
+  package:
+    name: "{{ suse_build_images_packages }}"
+    state: present
+  register: install_packages
+  until: install_packages is success
+  retries: 5
+  delay: 2
+  tags:
+    - install
+
+- name: Check if local docker registry server is running
+  shell: |
+    docker ps | grep {{ local_registry_name }}
+  register: registry_running_cmd_status
+  failed_when: false
+  tags:
+    - run
+
+- name: Start local docker registry if not running
+  shell: |
+    docker run -d -p {{ local_registry_port }}:{{ local_registry_port }} --name {{ local_registry_name }} --restart always registry:2
+  when:
+    - registry_running_cmd_status.rc != 0
+
+- name: Ensure local-registry DNS resolution is set in /etc/hosts
+  blockinfile:
+    path: /etc/hosts
+    block: |
+      {{ ansible_default_ipv4.address | ipaddr('address') }} {{ local_registry_name }}
+
+- name: Check docker daemon file is present on host
+  stat:
+    path: "/etc/docker/daemon.json"
+  register: _docker_daemon_conf_check
+
+- name: Make sure docker daemon file is present on host
+  file:
+    path: "/etc/docker/daemon.json"
+    state: touch
+    mode: 0644
+
+- name: Check if local-registry is already added in docker daemon conf file
```

Comment:
Alternatively, you could command cat/slurp the file here, and use a  `| from_json` to build a dict object on which you can check the presence of `insecure-registries` using a simple _docker_daemon_conf['insecure-registries']. It also will be easier to review/read ansible code when you'll set the value, and writing said value to file (`to_json`). The ordering doesn't matter if you write the file anyway, because you'll only write the file if there is a change to be made.

Reply:
Another alternative is to use quote filter to simplify reading.

Reply:
Good idea. Agree treating content as json and updating is defintely better. Will make change like that in next update.